src/shop-app.html

```diff
@@ -385,6 +388,10 @@
         });
       },
 
+      _onFallbackSelectionTriggered: function() {
+        this.page = '404';
+      },
+
```

Comment:
Can you move `_onFallbackSelectionTriggered` function after [_domChange](https://github.com/TimvdLippe/shop/blob/d04a803609c17a275d2e9fa8197064582b4391c0/src/shop-app.html#L544)? So all the handlers are together.